packages/mobile/src/screens/change-password-screen/ChangePasswordScreen.tsx

```diff
@@ -1,165 +1,189 @@
+
+const ChangePasswordHeaderLeft = ({ page }: { page: ChangePasswordPage }) => {
+  const navigation = useNavigation()
+  if (page === ChangePasswordPage.VerifyEmail) {
+    return <BackButton />
+  } else {
+    return (
+      <TouchableOpacity
+        onPress={() => navigation.navigate('AccountSettingsScreen')}
+      >
+        <IconClose size='l' color='subdued' />
+      </TouchableOpacity>
+    )
   }
+}
 
-  const handleNewPasswordSubmitted = (password: string) => {
-    dispatch(changePassword({ email, password, oldPassword }))
-  }
+const ChangePasswordNavigator = ({
+  page,
+  setPage
+}: {
+  page: ChangePasswordPage
+  setPage: (page: ChangePasswordPage) => void
+}) => {
+  const { handleSubmit, isSubmitting } = useFormikContext()
+  const styles = useStyles()
+  const navigation = useNavigation()
 
-  const handlePressDone = useCallback(() => {
-    navigation.goBack()
-    setCurrentPage(ChangePasswordPageStep.CONFIRM_CREDENTIALS)
-  }, [navigation, setCurrentPage])
+  // Only show the back button on the OTP page
+  const screenOptions = useAppScreenOptions({
+    headerRight: () => null,
+    headerLeft: () => {
+      return <ChangePasswordHeaderLeft page={page} />
+    }
+  })
 
+  // Map hook page state to screen navigations
   useEffect(() => {
-    if (changePasswordStatus === Status.LOADING) {
-      setCurrentPage(ChangePasswordPageStep.LOADING)
-    } else if (
-      currentPage === ChangePasswordPageStep.LOADING &&
-      changePasswordStatus === Status.SUCCESS
-    ) {
-      setCurrentPage(ChangePasswordPageStep.SUCCESS)
-    } else if (
-      currentPage === ChangePasswordPageStep.LOADING &&
-      changePasswordStatus === Status.ERROR
-    ) {
-      setCurrentPage(ChangePasswordPageStep.FAILURE)
-    }
-  }, [currentPage, setCurrentPage, changePasswordStatus])
-
-  const credentialsView = (
-    <View style={styles.contentContainer}>
-      <Text style={styles.header} variant='h1'>
-        {messages.changeHeader}
-      </Text>
-      <Text style={styles.text}>{messages.changeText}</Text>
-      <ConfirmCredentials onComplete={handleCredentialsConfirmed} />
-    </View>
+    navigation.navigate(ChangePasswordPage[page])
+  }, [page, navigation])
+
+  // Map navigations back to the hook page state
+  const stateListener = useCallback<
+    EventListenerCallback<EventMapCore<NavigationState>, 'state'>
+  >(
+    (e) => {
+      const state = e.data.state
+      const route = state.routes[state.index]
+      const newPage = ChangePasswordPage[route?.name]
+      if (newPage !== page) {
+        setPage(newPage)
+      }
+    },
+    [page, setPage]
   )
 
-  const loadingView = (
-    <LoadingSpinner style={{ alignSelf: 'center', marginVertical: 16 }} />
+  return (
+    <Screen variant='secondary' style={styles.screen}>
+      <ScreenContent>
+        <Stack.Navigator
+          screenOptions={screenOptions}
+          screenListeners={{
+            state: stateListener
```

Comment:
This was kind of annoying. I wonder if there's a better way than using effects for something like this to have subpagination/routing handled in a platform agnostic way?

Reply:
yeah maybe we can chat about what youre trying to do specifaclly? sign-up approach would likely solve what you need?

Reply:
would love to chat more!! basically tl;dr I want to control which page the user is on via the hook in common, but keep the ability to do `navigate`/`back` here. ie. the Android back button would take you back a screen etc and the hook needs to be aware you're back on a different "page"